src/rpc/util.cpp

```diff
@@ -277,8 +277,8 @@ CTxDestination AddAndGetMultisigDestination(const int required, const std::vecto
     if ((int)pubkeys.size() < required) {
         throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("not enough keys supplied (got %u keys, but need at least %d to redeem)", pubkeys.size(), required));
     }
-    if (pubkeys.size() > 16) {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "Number of keys involved in the multisignature address creation > 16\nReduce the number");
+    if (pubkeys.size() > MAX_PUBKEYS_PER_MULTISIG) {
```

Comment:
should we increase this amount? bitcoin increased it because they have segwit -> the transaction validation works faster for segwit txes. will it affect performance for us?

Reply:
The limit isn't actually going up, under the hood, the limit has always been up to 20 ([source](https://github.com/dashpay/dash/blob/19512988c6e6e8641245bd9c5fab21dd737561f0/src/script/interpreter.cpp#L1143-L1144), `MAX_PUBKEYS_PER_MULTISIG` has always been 20), the overhead involved with P2SH is what whittles that number down to 16 ([source](https://github.com/dashpay/dash/blob/19512988c6e6e8641245bd9c5fab21dd737561f0/src/script/descriptor.cpp#L877-L880)). Checking against 20 instead of 16 already takes place in `dash-tx` ([source](https://github.com/dashpay/dash/blob/19512988c6e6e8641245bd9c5fab21dd737561f0/src/bitcoin-tx.cpp#L333-L335)).

In effect, what the backport does is change the error from "too many public keys" (by testing against the without-overhead value as opposed to testing with) to "excessive redeemScript size" ([source](https://github.com/dashpay/dash/blob/19512988c6e6e8641245bd9c5fab21dd737561f0/src/rpc/util.cpp#L223-L225)) (which is the _actual_ upper bound when accounting for P2SH).

That is what "opens up" up to 20 pubkeys for multisig but SegWit but since we don't have that, it doesn't open up anything, just changes the error.